bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlComment.java

```diff
@@ -45,6 +46,28 @@ public XmlComment(String data, boolean readonly) {
         this.type = readonly ? PredefinedTypes.TYPE_READONLY_COMMENT : PredefinedTypes.TYPE_COMMENT;
     }
 
+    @Override
+    public IteratorValue getIterator() {
+        XmlComment that = this;
+        return new IteratorValue() {
+            boolean read = false;
+            @Override
+            public boolean hasNext() {
+                return !read;
+            }
+
+            @Override
+            public Object next() {
+                if (!read) {
+                    this.read = true;
+                    return that;
```

Comment:
Using `XmlComment.this` may work instead of introducing the new variable `that`.

Reply:
Seems like we'll have to use `"that"` here as well. It gives the same error as when testing with XmlText.
```
org.ballerinalang.core.util.exceptions.BLangRuntimeException: error: java.lang.ClassCastException {"message":"class io.ballerina.runtime.internal.values.XmlComment$1 cannot be cast to class io.ballerina.runtime.api.values.BXml (io.ballerina.runtime.internal.values.XmlComment$1 and io.ballerina.runtime.api.values.BXml are in unnamed module of loader 'app')"}
        at ballerina.lang.xml.0_8_0:externNext(internal.bal:35)
           ballerina.lang.xml.0_8_0.XMLIterator:next(internal.bal:31)
           xml_iteration:testXmlCommentSequenceIteration(xml_iteration.bal:195)
        at org.ballerinalang.test.BRunUtil.invoke(BRunUtil.java:440)
        at org.ballerinalang.test.BRunUtil.invokeFunction(BRunUtil.java:1198)
        at org.ballerinalang.test.BRunUtil.invoke(BRunUtil.java:188)
        at org.ballerinalang.test.BRunUtil.invoke(BRunUtil.java:1241)
        at org.ballerinalang.test.types.xml.XMLIterationTest.testXMLTypesForeachOp(XMLIterationTest.java:148)
```